### ISSUE TITLE
Exception accessing local-maven-repo-libs on windows paths

### ISSUE THREAD

**Issue by thorbolo** [spring-projects-experimental/spring-boot-migrator#867]
**Describe the bug**
The first Scan Process Aborts with the Exception below. Just start the Application with a given Path to a project

**SBM Version**
e8fcb8d47d898d86fcf24c25e2ed7f5e56cb0bae

**SBM Application**
SpringBootUpgradeReportApp

**To Reproduce**
Just start the Application with a given Path to a project. I am running it on a windows machine. My first guess would be there is something messed up with the path-Strings Slashes.
The omitted Libs it wants to retrieve from the local maven are company interns. So they are retrieved from within a pom.xml. 

**Expected behavior**
Getting a report :)

**Stacktrace**
07:51:56.435 [main] ERROR o.s.boot.SpringApplication - Application run failed
java.lang.IllegalStateException: Failed to execute ApplicationRunner
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:765)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:752)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1306)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1295)
	at org.springframework.sbm.SpringBootUpgradeReportApp.main(SpringBootUpgradeReportApp.java:31)
Caused by: java.lang.IllegalArgumentException: Illegal character in authority at index 7: file://C:\Users\<OMITTED>\.m2\repository/com/<....>
	at java.base/java.net.URI.create(URI.java:906)
	at org.openrewrite.maven.internal.MavenPomDownloader.download(MavenPomDownloader.java:487)
	at org.openrewrite.maven.tree.ResolvedPom$Resolver.resolveParentPropertiesAndRepositoriesRecursively(ResolvedPom.java:353)
	at org.openrewrite.maven.tree.ResolvedPom$Resolver.resolveParentPropertiesAndRepositoriesRecursively(ResolvedPom.java:364)
	at org.openrewrite.maven.tree.ResolvedPom$Resolver.resolveParentsRecursively(ResolvedPom.java:310)
	at org.openrewrite.maven.tree.ResolvedPom$Resolver.resolve(ResolvedPom.java:299)
	at org.openrewrite.maven.tree.ResolvedPom.resolve(ResolvedPom.java:144)
	at org.openrewrite.maven.tree.Pom.resolve(Pom.java:104)
	at org.openrewrite.maven.MavenParser.parseInputs(MavenParser.java:122)
	at org.springframework.sbm.build.impl.RewriteMavenParser.parseInputs(RewriteMavenParser.java:92)
	at org.springframework.sbm.project.parser.MavenProjectParser.parse(MavenProjectParser.java:94)
	at org.springframework.sbm.project.parser.ProjectContextInitializer.initProjectContext(ProjectContextInitializer.java:52)
	at org.springframework.sbm.engine.commands.ScanCommand.execute(ScanCommand.java:70)
	at org.springframework.sbm.SpringBootMigratorRunner.run(SpringBootMigratorRunner.java:48)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:762)
	... 5 common frames omitted
Caused by: java.net.URISyntaxException: Illegal character in authority at index 7: file://C:\Users\<OMITTED>\.m2\repository/com/<....>
	at java.base/java.net.URI$Parser.fail(URI.java:2974)
	at java.base/java.net.URI$Parser.parseAuthority(URI.java:3308)
	at java.base/java.net.URI$Parser.parseHierarchical(URI.java:3219)
	at java.base/java.net.URI$Parser.parse(URI.java:3175)
	at java.base/java.net.URI.<init>(URI.java:623)
	at java.base/java.net.URI.create(URI.java:904)
	... 19 common frames omitted

**Desktop (please complete the following information):**
 - OS: Windows 
 - Version 11
 - Oracle Java JDK 17.0.8

**Comment by thorbolo** [comment]
Oh but btw, love your project! Keep it up guys, this helps changing the worlds software to the better!

**Comment by fabapp2** [comment]
Hi @thorbolo
thanks for giving SBM a spin 🚀 
We started [reworking SBM](https://github.com/spring-projects-experimental/spring-boot-migrator/discussions/859).
The current (0.15.0) parser will be replaced with [this module](https://github.com/spring-projects-experimental/spring-boot-migrator/tree/main/sbm-support-rewrite).
Could you maybe try to scan your project with this parser once to see if it brings better results?

If you use [this test](https://github.com/spring-projects-experimental/spring-boot-migrator/blob/107c364ea70e1f22baf48b0385e43dc5c9f324aa/sbm-support-rewrite/src/test/java/org/springframework/sbm/parsers/RewriteMavenProjectParserIntegrationTest.java#L63)

You'd just need to change the `baseDir`; if the parsing passes, you'll have more luck with the next release.

**Comment by fabapp2** [comment]
I'll close this issue for now, hoping it works with the new parser 🤞 
If not, please open a new issue.

**Comment by praveen19bits** [comment]
@fabapp2 can you let me know how to resolve this exception? I am also receiving the same exception. I was trying to upgrade spring boot version from 2.x.x to 3.x.x using the given JAR.

**Comment by fabapp2** [comment]
Hi @praveen19bits. Are you relying on .m2/settings.xml to build your project

**Comment by praveen19bits** [comment]
@fabapp2 Thanks for your reply.  Yes we are using .m2/settings.xml to build the project.

 To use the new parser, what exactly do I need to do? Should I take this repo and build and use that jar to upgrade SB version? Or Can I do something with spring-boot-upgrade.jar  to pass arguments?

I did not understand your comment above. If you can elaborate your solution, will help me.

**Comment by javabean68** [comment]
Hi @fabapp2 

I get this exception too: 
`aused by: java.net.URISyntaxException: Illegal character in authority at index 7: file://C:\Users<OMITTED>.m2\repository/com/<....>
at java.base/java.net.URI$Parser.fail(URI.java:2974)
at java.base/java.net.URI$Parser.parseAuthority(URI.java:3308)
at java.base/java.net.URI$Parser.parseHierarchical(URI.java:3219)`

If I understood a new release should address this problem? If so, when is it due? 
Thank you and compliments on your work: it seems really interesting!
Bye
Fabio

**Comment by fabapp2** [comment]
> @fabapp2 Thanks for your reply. Yes we are using .m2/settings.xml to build the project.
> 
> To use the new parser, what exactly do I need to do? Should I take this repo and build and use that jar to upgrade SB version? Or Can I do something with spring-boot-upgrade.jar to pass arguments?
> 
> I did not understand your comment above. If you can elaborate your solution, will help me.

The new parser is part of a larger revamp, see #859 
This means that the existing modules (e.g. recipes) will be removed or migrated to the new parser. 
If you need to parse a project with settings.xml you can use `RewriteMavenProjectParser`, see [README](https://github.com/spring-projects-experimental/spring-boot-migrator/blob/main/sbm-support-rewrite/README.adoc).
This will parse your project to OpenRewrite AST and you can run OR recipes against it.
Otherwise, you'll have to wait for the revamp to be complete.

**Comment by durcon** [comment]
@fabapp2 

I don't think that this is a problem of the `settings.xml`. I removed it, but I get still the same error:

```
Caused by: java.net.URISyntaxException: Illegal character in authority at index 7: file://C:\Users\dur\.m2\repository/org/springframework/boot/spring-boot-starter-parent/2.7.15/spring-boot-starter-parent-2.7.15.pom
        at java.base/java.net.URI$Parser.fail(URI.java:2974)
        at java.base/java.net.URI$Parser.parseAuthority(URI.java:3308)
        at java.base/java.net.URI$Parser.parseHierarchical(URI.java:3219)
        at java.base/java.net.URI$Parser.parse(URI.java:3175)
        at java.base/java.net.URI.<init>(URI.java:623)
        at java.base/java.net.URI.create(URI.java:904)
        ... 26 more
```

The problem seems to be the windows path. At index 7 is the letter of the windows drive.

**Comment by fabapp2** [comment]
Interesting. I will look into it.

**Comment by fabapp2** [comment]
The issue should be fixed with #942.
Thank you all for your help and investigations!
Creating a new release now.
